src/app/pages/dashboard/components/widget-group-form/widget-group-slot-form/widget-group-slot-form.component.ts

```diff
@@ -148,6 +164,27 @@ export class WidgetGroupSlotFormComponent implements AfterViewInit, OnChanges {
     this.setValuesFromInput();
   }
 
+  ngOnInit(): void {
+    runInInjectionContext(this.environmentInjector, () => {
```

Comment:
Large and complicated function needs to be split into methods/utility functions. Perhaps you can also figure out a way to extract similar code here and in `widget-group` into a shared function.